server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPServerTest.java

```diff
@@ -232,6 +232,28 @@ void serverShouldKeepReceivingErrorResponseWhenAlways() throws Exception {
                 .isInstanceOf(SMTPConnectionClosedException.class)
                 .hasMessageContaining("421");
         }
+
+        @Test
+        void serverShouldDecreaseNumberOfAnswerAfterMatched() throws Exception {
+            mockServer.setBehaviors(new MockSMTPBehavior(
+                MAIL_FROM,
+                Optional.empty(),
+                Response.serverReject(SERVICE_NOT_AVAILABLE_421, "mock response"),
+                MockSMTPBehavior.NumberOfAnswersPolicy.times(5)));
+
+            try {
+                smtpClient.sendMessage(mail1);
+            } catch (SMTPConnectionClosedException e) {
+                // ignore first failure;
+            }
+
+            assertThat(mockServer.getBehaviorRepository().allBehaviors())
+                .containsOnly(new MockSMTPBehavior(
+                    MAIL_FROM,
+                    Optional.empty(),
+                    Response.serverReject(SERVICE_NOT_AVAILABLE_421, "mock response"),
+                    MockSMTPBehavior.NumberOfAnswersPolicy.times(4)));
```

Comment:
Mmmh IMO that is the wrong model to be using.

We should keep the immutable "total" number of answer (setted up by our policy) immutable.

We could also add a mutable "answerCount" that would then allow tracking if that behavior should still be applied or not.